src/io/io_behaviorsaver.js

```diff
@@ -58,13 +58,89 @@ IO.BehaviorSaver = new (function() {
 		}
 	}
 
+    String.prototype.format = function() {
+        var formatted = this;
+        for( var arg in arguments ) {
+            formatted = formatted.replace("{" + arg + "}", arguments[arg]);
+        }
+        return formatted;
+    };
+
 	var saveSuccessCallback = function() {
 		T.logInfo("Save successful!");
 		UI.Panels.Terminal.hide();
 		UI.Settings.updateBehaviorlib();
 		UI.Tools.notifyRosCommand('save');
-	}
+    }
+
+    var saveOnRobot = function() {
+        var names = Behavior.createNames();
+        var package_name = names.rosnode_name;
+        var ssh_username = UI.Settings.getRobotUsername();
+        var ssh_hostname = UI.Settings.getRobotHostname();
+        var ssh_package_path = ""
+
+        if(ssh_username === "" || ssh_hostname === "" /*|| ssh_password === ""*/) {
+            T.logError("SSH configurations are not well defined, fix them from Configuration Tab -> Code Generation Box");
+            return;
+        }
+
+        var ssh_password = window.prompt("Please enter {0}:{1} password: ".format(ssh_username, ssh_hostname));
+        if(ssh_password == null || ssh_password == "") {
+            return;
+        }
 
+        ROS.getPackagePath(package_name, (package_path) =>
+        {
+            try
+            {
+                var SSHClient = require('ssh2').Client;
+                conn = new SSHClient();
+                conn.on('ready', function() {
+                  var ros_distro = process.env.ROS_DISTRO;
+                  conn.exec('bash -c "source .profile && rospack find {1}"'.format(ros_distro, package_name), function(err, stream) {
+                    if (err) {
+                        T.logError(err);
+                    }
+                    stream.on('close', function(code, signal) {
+                      conn.end();
+                    })
+                    .on('data', function(remote_package_path) {
+                      ssh_package_path = String(remote_package_path).trim();
+                      T.logInfo('Path to package on the robot: ' + ssh_package_path);
+                      conn.end()
+                      var scp_client = require('scp2')
+                      var ssh_query = '{0}:{1}@{2}:{3}/'.format(
+                          ssh_username,
+                          ssh_password,
+                          ssh_hostname,
+                          ssh_package_path);
```

Comment:
This contains the password in plain text, right? Is there a way to encrypt? I have no experience with the scp2 package, but do you know how the query string would be further used?